src/object/object_domain.c

```diff
@@ -2683,7 +2686,8 @@ tp_domain_find_charbit (DB_TYPE type, int codeset, int collation_id, unsigned ch
 		{
 		  break;	/* found */
 		}
-	      else if (dom->collation_id == collation_id && dom->collation_flag == collation_flag)
+	      else if (dom->collation_id == collation_id && dom->collation_flag == collation_flag
+		       && dom->codeset == codeset)
```

Comment:
I have the same question as the one I asked above.

Reply:
OK, good point. I fixed it for CHAR and NCHAR.